Sources/NIO/FileHandle.swift

```diff
@@ -81,12 +81,56 @@ public final class NIOFileHandle: FileDescriptor {
 }
 
 extension NIOFileHandle {
+    /// `Mode` represents file access modes.
+    public struct Mode: OptionSet {
+        public let rawValue: UInt8
+
+        public init(rawValue: UInt8) {
```

Comment:
Is there any reason for this initialiser to be public?

Reply:
Yes, it is a requirement of `OptionSet`

Reply:
Wow I always forget how annoying a protocol `OptionSet` is.